File: app/src/main/java/com/talktiva/pilot/helper/NetworkChangeReceiver.java
package com.talktiva.pilot.helper;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkChangeReceiver extends BroadcastReceiver {

    private Utility utility;

    public NetworkChangeReceiver() {
    }

    public NetworkChangeReceiver(Activity activity) {
        utility = new Utility(activity);
    }

    @SuppressLint("UnsafeProtectedBroadcastReceiver")
    @Override
    public void onReceive(Context context, Intent intent) {
        ConnectivityManager manager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo wifi = manager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
        NetworkInfo mobile = manager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);
        if (wifi.isAvailable() || mobile.isAvailable()) {
            if (!wifi.isConnected() && !mobile.isConnected()) {
                utility.requestInternet();
            } else {
                utility.dismissRequestInternet();
            }
        }
    }
}

File: app/src/main/java/com/talktiva/pilot/helper/NetworkChangeReceiver.kt
package com.talktiva.pilot.helper

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.net.NetworkInfo

@Suppress("DEPRECATION")
class NetworkChangeReceiver : BroadcastReceiver() {

    @SuppressLint("UnsafeProtectedBroadcastReceiver")
    override fun onReceive(context: Context, intent: Intent) {
        val manager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val wifi: NetworkInfo? = manager.getNetworkInfo(ConnectivityManager.TYPE_WIFI)
        val mobile: NetworkInfo? = manager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE)
        if ((wifi != null && wifi.isAvailable) || (mobile != null && mobile.isAvailable)) {
            if (!wifi!!.isConnected && !mobile!!.isConnected) {
                Utility.requestInternet(context)
            } else {
                Utility.dismissRequestInternet()
            }
        }
    }
}
